bootloader/bootloader.c

```diff
@@ -194,9 +199,11 @@ int main()
 	#endif
 #endif
 
-#ifdef USB_PIN_DPU
+#if defined(USB_PIN_DPU) && !defined(USB_DPU_PORT)
 	// This drives USB_PIN_DPU (D- Pull-Up) high, which will tell the host that we are going on-bus.
 	LOCAL_EXP(GPIO,USB_PORT)->BSHR = 1<<USB_PIN_DPU;
+#else
+	LOCAL_EXP(GPIO,USB_DPU_PORT)->BSHR = 1<<USB_PIN_DPU;
 #endif
```

Comment:
Keep in mind that `USB_PIN_DPU` is entirely optional. The `#else` will break this. I guess it should just be two separate ifs `#if defined(USB_PIN_DPU) && !defined(USB_DPU_PORT)` and `#if defined(USB_PIN_DPU) && defined(USB_DPU_PORT)`

Reply:
Thanks for catching that. I've changed it as suggested.